### PR TITLE
Expand functional test scripts

### PR DESCRIPTION
**Pull request by betaredex**
Added several new tests:
offer not available in time range test
offer claim multiple lessees test
lease create test
offer create invalid user test
lease create invalid user test
offer list expired offer test
offer claim expired offer test
lease list expired lease test

### CODE REVIEW

File: tests/functional/test-lease-expire.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  -f shell > $tmpfile \
+  || { ec=$?; echo "ERROR: failed to claim offer" >&2; exit $ec; }
+
+echo "OFFER CLAIM TEST SUCCEEDED"
+
+sleep 2m
+date +"%Y-%m-%D %H:%M:%S"
+
+###################################
+## LEASE LIST EXPIRED OFFER TEST ##
```
Comment (DanNiESh) [inline tests/functional/test-lease-expire.sh:39]: I'm a little confused by the test name. It's a lease expired not an offer expired test, right?

File: tests/functional/test-lease-expire.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+###################################
+
+echo "OFFER LIST EXPIRED OFFER TEST"
```
Comment (DanNiESh) [inline tests/functional/test-lease-expire.sh:42]: "LEASE LIST EXPIRED OFFER TEST"  I guess?

File: tests/functional/test-no-offer.sh [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+echo "LEASE DELETE TEST"
+
+openstack  --os-cloud test1 esi lease delete $uuid \
+  || { ec=$?; echo "ERROR: failed to delete lease" >&2; exit $ec; }
+
+echo "LEASE DELETE TEST SUCCEEDED"
+
+####################################
+## OFFER CREATE INVALID USER TEST ##
```
Comment (DanNiESh) [inline tests/functional/test-no-offer.sh:65]: This test requires an offer to be created, do you think it could be placed in test-no-lessee-id.sh?

**Review by DanNiESh — COMMENTED** [review]
Looks good overall! I left some comments on minor issues.

File: tests/functional/test-lease-expire.sh (hunk 1 above)
Comment (betaredex) [reply tests/functional/test-lease-expire.sh:39]: copy pasted code lol
will fix

File: tests/functional/test-no-offer.sh (hunk 3 above)
Comment (betaredex) [reply tests/functional/test-no-offer.sh:65]: This doesn't require an offer to be created, it creates an offer. If I put it in the other script it'd have a time conflict with the offer in that script and might not give the error I want.

Comment (DanNiESh) [reply tests/functional/test-no-offer.sh:65]: I see!

**Comment by betaredex** [comment]
Fixed typo

File: tests/functional/test-lease-expire.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+openstack --os-cloud test1-subproject esi offer claim $uuid \
+  --start-time="$start" \
+  --end-time="$(date -d "+1 minute" +"%Y-%m-%d %H:%M:%S")" \
+  -f shell > $tmpfile \
+  || { ec=$?; echo "ERROR: failed to claim offer" >&2; exit $ec; }
+
+echo "OFFER CLAIM TEST SUCCEEDED"
+
+sleep 2m
```
Comment (tzumainn) [inline tests/functional/test-lease-expire.sh:35]: You may want to print out a statement that the test will pause for 2 minutes, just so someone running the test knows what to expect.

File: tests/functional/test-no-offer.sh [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    "server_config": {
+        "new attribute XYZ": "This is just a sample list of free-form attributes used for describing a server.",
+        "cpu_type": "Intel Xeon",
+        "cores": 16,
+        "ram_gb": 512,
+        "storage_type": "samsung SSD",
+        "storage_size_gb": 204
+    }
+}
+EOF
```
Comment (tzumainn) [inline tests/functional/test-no-offer.sh:27]: Is it possible to genericize setup.sh somehow to cover this node/temp file creation?

File: README.md [hunk 6]
```diff
@@ -84,10 +84,11 @@ Our code development features a mix of upstream OpenStack work and custom ESI co
 
 ## Contact and Contribution
 
-You can contact the ESI development team on the Freenode `#moc` IRC channel.
+You can contact the ESI development team on the [OFTC][oftc] `#moc` IRC channel.
```
Comment (DanNiESh) [inline README.md:87]: Why is this commit included in your PR?

Comment (betaredex) [reply README.md:87]: I am bad at git
I was trying to fix a merge conflict but I guess I messed it up

File: tests/functional/test-no-offer.sh (hunk 5 above)
Comment (betaredex) [reply tests/functional/test-no-offer.sh:27]: setup.sh creates an offer as part of the script, so I didn't want to use it for the no-offer script
I didn't expect I'd need any other scripts that don't create an offer when they're being set up, so I didn't think I should move this to its own script

File: tests/functional/test-lease-expire.sh (hunk 4 above)
Comment (betaredex) [reply tests/functional/test-lease-expire.sh:35]: Sounds good

File: tests/functional/test-no-offer.sh (hunk 5 above)
Comment (tzumainn) [reply tests/functional/test-no-offer.sh:27]: okay! I think this could probably be restructured somewhat, but that can be done later - it's fine for now!

**Review by tzumainn — APPROVED** [review]
Looks good - thanks!

**Pull request merged**